aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -58,199 +59,206 @@ namespace {
+            TORCH_CHECK(end - start <= nplain_dims,
+                        compressed_indices_name, "[", batch_indices, i, "] (=", end, ") - ",
+                        compressed_indices_name, "[", batch_indices, i - 1, "] (=", start, ") <= number of ",
+                        plain_dim_name, "s (=", nplain_dims, ") is unsatisfied", at_batch_id);
+            int64_t last_plain_index = -1;
+            for (int n = start; n < end; n++) {
+              int64_t plain_index = plain_indices_data_ptr[batch_id*batch_plain_stride + n];
+              // 5.4, 5.5
+              TORCH_CHECK(0 <= plain_index && plain_index < nplain_dims,
+                          plain_indices_name, "[", batch_indices, n, "] (=", plain_index, ") is out of range (0, ", nplain_dims, ")", at_batch_id);
+              // 5.6
+              TORCH_CHECK(plain_index > last_plain_index, plain_indices_name, " must be ordered sequence of distinct integers but ",
+                          plain_indices_name, "[", batch_indices, n - 1, "] (=", last_plain_index, ") < ",
+                          plain_indices_name, "[", batch_indices, n, "] (=", plain_index, ") is unsatisfied", at_batch_id);
+              last_plain_index = plain_index;
+            }
+            start = end;
           }
         }
-        if (plain_indices.numel() > 0) {
-          TORCH_CHECK(0 <= plain_indices.min().item<index_t>(), plain_indices_name, ".min() should be greater or equal to zero");
-          TORCH_CHECK(size[plain_dim] > plain_indices.max().item<index_t>(), "size[-", (size.size() - plain_dim),"] should be greater than ", plain_indices_name, ".max()");
-        }
-      });
+      }
+    });
 
   // Device Invariants
-  TORCH_CHECK(
-      plain_indices.get_device() == compressed_indices.get_device(),
-      compressed_indices_name, " and ", plain_indices_name, " devices (",
-      compressed_indices.get_device(),
-      ", ",
-      plain_indices.get_device(),
-      ") must match");
-  TORCH_CHECK(
-      compressed_indices.get_device() == values.get_device(),
-      "device of ", compressed_indices_name, " (",
-      compressed_indices.get_device(),
-      ") must match device of values (",
-      values.get_device(),
-      ")");
+  // 4.1
   TORCH_CHECK(
       values.device().type() == kCPU || values.device().type() == kCUDA,
       "device type of values (",
       values.device().type(),
       ") must be CPU or CUDA");
-
+  // 4.2, 4.3, 4.4
+  TORCH_CHECK(
+      compressed_indices.get_device() == values.get_device(),
+      "device of ", compressed_indices_name, " (=",
+      compressed_indices.device(),
+      ") must match device of values (=",
+      values.device(),
+      ")");
+  TORCH_CHECK(
+      compressed_indices.get_device() == plain_indices.get_device(),
+      "device of ", compressed_indices_name, " (=",
+      compressed_indices.device(),
+      ") must match device of ", plain_indices_name," (=",
+      plain_indices.device(),
+      ")");
 }
```

Comment:
A helper function could be used here to eliminate duplicates.

Reply:
I am not sure if a helper function is worth implementing here considering that it would be the only use case of such a function and the outputs of these checks are fully tested in
https://github.com/pytorch/pytorch/blob/384ae783c478040bd9ec5142e648786b10598a52/test/test_sparse_csr.py#L741-L759
What do you think, @nikitaved?

Reply:
Makes sense. Fine by me :)